recipes/libelf/all/conanfile.py

```diff
@@ -41,7 +40,10 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
-        if self.options.shared:
+        if self.settings.os not in ["Linux", "FreeBSD", "Windows"]:
+            self.options.rm_safe("shared")
+            self.package_type = "static-library"
```

Comment:
shouldn't it be done in config_options()?

Reply:
absolutely!